cdap-explore/src/test/java/co/cask/cdap/explore/service/HiveExploreServiceStreamTest.java

```diff
@@ -108,7 +134,26 @@ public void testStreamDefaultSchema() throws Exception {
   }
 
   @Test
-  public void testSelectStarOnStream() throws Exception {
+  public void testStreamAuthorization() throws Exception {
+    StreamId streamId = NAMESPACE_ID.toEntityId().stream(streamName);
+    revokeAndAssertSuccess(streamId, USER, ImmutableSet.of(Action.ALL));
+    grantAndAssertSuccess(streamId, USER, ImmutableSet.of(Action.ADMIN, Action.WRITE));
+    // without READ privilege, explore test should fail
+    try {
+      testSelectStarOnStream();
+    } catch (Exception e) {
+      Assert.assertTrue(e.getCause() instanceof ExploreException);
+    }
+
+    // now grant READ privilege and the test should pass
+    revokeAndAssertSuccess(streamId, USER, ImmutableSet.of(Action.ADMIN, Action.WRITE));
+    grantAndAssertSuccess(streamId, USER, ImmutableSet.of(Action.READ));
+    testSelectStarOnStream();
```

Comment:
@bdmogal I am not sure about datasets authorization in explore. @chtyim Do we have tests that tests dataset authorization in explore (both select and insert) ?